doc/ref/states/writing.rst

```diff
@@ -153,7 +153,11 @@ A State Module must return a dict containing the following keys/values:
       However, if a state is going to fail and this can be determined
       in test mode without applying the change, ``False`` can be returned.
 
-- **comment:** A string containing a summary of the result.
+- **comment:** A list of strings (or a single string) summarizing the result.
+  Lists of strings will be joined with newlines to form the final comment;
+  this is useful to allow multiple comments from subparts of a state.
+  Prefer to keep line lengths short (use multiple lines as needed),
+  and end with punctuation (e.g. a period) to delimit multiple comments.
```

Comment:
We should probably indicate when this change happened by adding something like `As of Salt Oxygen, ...` or something along those lines.